doc/how_to_guides/pilz_industrial_motion_planner/pilz_industrial_motion_planner.rst

```diff
@@ -397,13 +397,199 @@ constraints of an individual ``moveit_msgs::msg::MotionPlanRequest`` are
+       RCLCPP_ERROR(LOGGER, "Error waiting for action server /sequence_move_group");
+       return -1;
+    }
+  
+Then, the request is created:
+
+::
+   
+    // Create a MotionSequenceRequest
+    moveit_msgs::msg::MotionSequenceRequest sequence_request;
+    sequence_request.items.push_back(item1);
+    sequence_request.items.push_back(item2);
+
+Create goal and planning options. A goal response callback and result callback can be included as well.
+
+::
+
+    // Create action goal
+    auto goal_msg = MoveGroupSequence::Goal();
+    goal_msg.request = sequence_request;
+ 
+    // Planning options
+    goal_msg.planning_options.planning_scene_diff.is_diff = true;
+    goal_msg.planning_options.planning_scene_diff.robot_state.is_diff = true;
+    // goal_msg.planning_options.plan_only = true;
+
+Finally, send the goal request and wait for the response:
+
+::
+
+    // Send the action goal
+    auto goal_handle_future = client->async_send_goal(goal_msg, send_goal_options);
+ 
+    // Get result
+    auto future_result = client->async_get_result(goal_handle_future.get());
+ 
+    // Wait for the result
+    if (future_result.valid()) {
+       auto result = future_result.get();  // Blocks the program execution until it gets a response
+       RCLCPP_INFO(LOGGER, "Action completed. Result: %d",  static_cast<int>(result.code));
+    } else {
+       RCLCPP_ERROR(LOGGER, "Action couldn't be completed.");
+    }
+
+If the motion need to be stopped mid trajectory, the action have to be cancel with:
+
+::
+
+    auto future_cancel_motion = client->async_cancel_goal(goal_handle_future_new.get());
+
+
+The ``pilz_robot_programming`` package for a `ROS 1 Python script
 <https://github.com/PilzDE/pilz_industrial_motion/blob/melodic-devel/pilz_robot_programming/examples/demo_program.py>`_
-that shows how to use the capability.
+shows how to use the capability.
 
 Service interface
 ~~~~~~~~~~~~~~~~~
 
 The service ``plan_sequence_path`` allows the user to generate a joint
 trajectory for a ``moveit_msgs::msg::MotionSequenceRequest``.
 The trajectory is returned and not executed.
+
+The service client needs to be initialize:
+
+::
+
+    // MoveGroupSequence service client
+    using GetMotionSequence = moveit_msgs::srv::GetMotionSequence;
+    auto service_client = node->create_client<GetMotionSequence>("/plan_sequence_path");
+ 
+    // Verify that the action server is up and running
+    while (!service_client->wait_for_service(std::chrono::seconds(10))) {
+       RCLCPP_WARN(LOGGER, "Waiting for service /plan_sequence_path to be available...");
+    }
+
+Then, the request is created:
+
+::
+
+    // Create request
+    auto service_request = std::make_shared<GetMotionSequence::Request>();
+    service_request->request.items.push_back(item1);
+    service_request->request.items.push_back(item2);
+
+Service call and response. The method ``future.get()`` blocks the execution of the program until the server response arrives.
```

Comment:
Actually, `future.get()` will not block and wait for a response, which is maybe what you were running into!

You should be doing something like this:
https://docs.ros.org/en/humble/Tutorials/Beginner-Client-Libraries/Writing-A-Simple-Cpp-Service-And-Client.html#write-the-client-node

Reply:
That was just a comment! It is not a issue, but it is how the service works, as long as I've seen. In the Pilz Industrial Motion Planner package, the class MoveGroupSequenceService of the [move_group_sequence_service.h](https://github.com/moveit/moveit2/blob/main/moveit_planners/pilz_industrial_motion_planner/include/pilz_industrial_motion_planner/move_group_sequence_service.h) only has a **plan** method. On the other hand, the MoveGroupSequenceAction of the [move_group_sequence_action.h](https://github.com/moveit/moveit2/blob/main/moveit_planners/pilz_industrial_motion_planner/include/pilz_industrial_motion_planner/move_group_sequence_action.h) has plan and execution methods.

I didn´t use the rclcpp::spin_until_future_complete since the node is already spinning, set by the executor at the beginning of the code. That gave me some error when compiling. However, this is just necessary for the visuals, so we could forget about the trajectory representation and do it the right way.

Reply:
If the node is already spinning elsewhere, then you likely just want to do a `future.wait_for(1s)` or something like that, and then check its status:

https://en.cppreference.com/w/cpp/thread/future/wait_for

See, for example, line 152 of [this link](https://docs.ros2.org/latest/api/rclcpp/static__single__threaded__executor_8hpp_source.html), but then use a non-zero timeout.

Reply:
If the node is already spinning elsewhere, then you likely just want to do a `future.wait_for(1s)` or something like that, and then check its status:

https://en.cppreference.com/w/cpp/thread/future/wait_for

See, for example, line 152 of [this link](https://docs.ros2.org/latest/api/rclcpp/static__single__threaded__executor_8hpp_source.html), but then use a non-zero timeout.

Reply:
I tried it and it works, but if the timeout is too low it doesn´t work, since it returns before the actions (or service) is finished. The [wait_for documentation](https://en.cppreference.com/w/cpp/thread/shared_future/wait_for) shows an example using a while loop to use multiple future.wait_for(1s) until the goal is reached. 

```
do {
  switch (action_status = future_result.wait_for(std::chrono::seconds(1)); action_status)
  {
    case std::future_status::deferred:
      std::cout << "Deferred\n";
      break;
    case std::future_status::timeout:
      std::cout << "Executing trajectory...\n";
      break;
    case std::future_status::ready:
      std::cout << "Action ready!\n";
      break;
  }    
}
while (action_status != std::future_status::ready);
```
This works fine and we can have the RVIZ visuals at the same time. I should change the std::cout with the log.

In the tutorial I could write how to use the rclcpp::spin_until_future_complete as well.